### PR TITLE
Add game ending logic

### PR DESCRIPTION
**Pull request by duouyang**
Server will check whether the game should end in each tick. If the game ends, all four client threads will exit with status 0.
Also, if a player is dead, then the client stop to send clientInput to server. And the player's position, attack ... will not be considered on the server side.

### CODE REVIEW

File: client/Window.h.orig [hunk 1]
```diff
@@ -5,10 +5,7 @@
 #include "shader.h"
 #include "Character.h"
 #include "EnvElement.h"
-<<<<<<< HEAD
 //#include "../common/constants.h"
-=======
->>>>>>> main
```
Comment (duouyang) [inline client/Window.h.orig:11]: This parts seem to be commented in the main thread before merging, so I just left it commented here

Comment (duouyang) [reply client/Window.h.orig:11]: Same for the following files

File: common/game/Game.cpp [hunk 2]
```diff
@@ -368,7 +368,7 @@ void projectileMove(Projectile* p) {
     2. outside of the map
     3. hit the obstacle
 
-    Note: isEnd will not check any collision
```
Comment (siueric0010) [inline common/game/Game.cpp:371]: i assume you mean projectileIsEnd will not check any collisions, not checkEnd.

File: common/game/GamePlayer.h [hunk 3]
```diff
@@ -23,7 +23,7 @@
 
 
 class GamePlayer {
-protected:
```
Comment (siueric0010) [inline common/game/GamePlayer.h]: im fine with this but i didn't see any direct field accesses so idk if you actually need this to be public

**Review by siueric0010 — APPROVED** [review]
i approve for now, idk if you want to do some changes to some things before you merge, so I'll leave it up to you

**Pull request merged**